api/handlers.go

```diff
@@ -31,3 +32,123 @@ func (api *API) userHandlerGET(w http.ResponseWriter, req *http.Request, _ httpr
 	}
 	WriteJSON(w, u)
 }
+
+// userUploadsHandler returns all uploads made by the current user.
+func (api *API) userUploadsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
```

Comment:
I'm not sure why we would not just add pagination from the start. It's just two query string parameters `offset` (or `skip`) and `limit`. Pagination is a pretty basic luxury to have when serving large JSON responses.

Reply:
Added.